### DISCUSSION TITLE
Need help with Dependabot and Github Actions

### DISCUSSION THREAD

**Discussion by dihmeetree** [community/community#26065] [Actions]
<p>What i’m looking to do is that when Dependabot creates PR’s to update dependencies, it will do the following:</p>
<ol>
<li>
<p><strong>Run the NodeCI workflow defined as such</strong></p>
<p>name: Node CI</p>
<p>on:<br>
pull_request:<br>
types: [opened, synchronize, ready_for_review, review_requested, reopened]<br>
branches:<br>
- master</p>
<p>jobs:<br>
build:<br>
name: Build with Node.js<br>
if: github.base_ref == ‘master’ &amp;&amp; (github.actor == ‘dependabot[bot]’ || github.actor == ‘dependabot-preview[bot]’)<br>
runs-on: ubuntu-latest<br>
timeout-minutes: 10<br>
strategy:<br>
matrix:<br>
node-version: [8.x, 10.x, 12.x]<br>
steps:<br>
- name: Dump GitHub context<br>
env:<br>
GITHUB_CONTEXT: ${{ toJson(github) }}<br>
run: echo “$GITHUB_CONTEXT”</p>
<pre><code>   - name: Checkout repo
     uses: actions/checkout@v1

   - name: Use Node.js ${{ matrix.node-version }}
     uses: actions/setup-node@v1
     with:
       node-version: ${{ matrix.node-version }}

   - name: NPM install, build, and test
     run: |
       npm ci
       npm run build --if-present
       npm test
     env:
       CI: true
</code></pre>
<p>approve:<br>
name: Approve pull request<br>
runs-on: ubuntu-latest<br>
timeout-minutes: 10<br>
needs: build<br>
steps:<br>
- uses: hmarr/auto-approve-action@v2.0.0<br>
with:<br>
github-token: ‘${{ secrets.GITHUB_TOKEN }}’</p>
<p>merge:<br>
name: Merge pull request<br>
runs-on: ubuntu-latest<br>
timeout-minutes: 10<br>
needs: approve<br>
steps:<br>
- name: Merge<br>
uses: actions/github-script@0.8.0<br>
with:<br>
script: |<br>
github.pulls.merge({<br>
owner: context.payload.repository.owner.login,<br>
repo: context.payload.repository.name,<br>
pull_number: context.payload.pull_request.number<br>
})<br>
github-token: ‘${{ secrets.GITHUB_TOKEN }}’</p>
</li>
<li>
<p><strong>Once the NodeCI Workflow approves the PR (bypasses the need to have a reviewer approve it), and merges the PR, I have another workflow, that deploys the PR to Cloudflare Workers. That Deploy workflow look’s like the following.</strong></p>
<p>name: Deploy to Cloudflare Workers</p>
<p>on:<br>
pull_request:<br>
types: [closed]<br>
branches:<br>
- master</p>
<p>env:<br>
NODE_VERSION: 12.x</p>
<p>jobs:<br>
deploy:<br>
name: Deploy<br>
runs-on: ubuntu-latest<br>
timeout-minutes: 10<br>
steps:<br>
- name: Dump GitHub context<br>
env:<br>
GITHUB_CONTEXT: ${{ toJson(github) }}<br>
run: echo “$GITHUB_CONTEXT”</p>
<pre><code>   - name: Checkout repo
     uses: actions/checkout@master

   - name: Setup Node.js ${{ env.NODE_VERSION }}
     uses: actions/setup-node@v1.4.0
     with:
       node-version: ${{ env.NODE_VERSION }}

   - name: Cache node modules
     uses: actions/cache@v1
     env:
       cache-name: cache-node-modules
     with:
       path: ~/.npm
       key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
       restore-keys: |
         ${{ runner.os }}-build-${{ env.cache-name }}-
         ${{ runner.os }}-build-
         ${{ runner.os }}-

   - name: Install NextJS dependencies
     run: npm ci

   - name: Build NextJS application
     run: npm run build

   - name: Export NextJS application
     run: npm run export

   - name: Publish to Cloudflare Workers
     uses: cloudflare/wrangler-action@1.1.0
     if: github.event_name == 'closed'
     with:
       apiToken: ${{ secrets.CF_API_TOKEN }}
       environment: 'production'

   - name: Cleanup @cloudflare/wrangler data
     continue-on-error: true
     run: sudo rm -rf ~/.wrangler .wrangler ~/.config .config ~/configstore configstore
</code></pre>
</li>
</ol>
<p><strong>I assume that after the NodeCI workflow merges (closes) the PR… that because I defined:</strong></p>
<pre><code>on:
  pull_request:
    types: [closed]
    branches:
      - master
</code></pre>
<p><strong>It will run my Deploy workflow? </strong></p>
<p><strong>However the issue is that it is not. It runs the NodeCI workflow just fine, however the Deploy workflow is never triggered, even after the Dependabot PR is approved/merged. Does anyone have any ideas what i’m doing wrong?</strong></p>

**Comment by dihmeetree** [comment, marked as answer]
<p>Solved by replacing <strong>secrets.GITHUB_TOKEN</strong> with a Github Personal Access Token (<a href="https://github.com/settings/tokens" rel="nofollow noopener">https://github.com/settings/tokens</a>), on the merge step!</p>
<pre><code>merge:
    name: Merge pull request
    runs-on: ubuntu-latest
    timeout-minutes: 10
    needs: approve
    steps:
      - name: Merge
        uses: actions/github-script@0.8.0
        with:
          script: |
            github.pulls.merge({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.pull_request.number
            })
          github-token: '${{ **secrets.GITHUB\_PERSONAL\_TOKEN** }}'
</code></pre>

**Comment by lannonbr** [comment]
<p>For people who come upon this in the future, it is due to that at the time of writing this, one workflow can not trigger another workflow, so by changing from the default github token that is provided to a personal access  token.</p>
